pkg/discovery/worker/compliance/unschedulable_node_anti_affinity_group_dto_builder.go

```diff
@@ -0,0 +1,100 @@
+package compliance
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	"github.com/turbonomic/turbo-go-sdk/pkg/builder/group"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+)
+
+// In Turbo, Schedulable nodes are marked with an access commodity 'schedulable'. Pods are required to buy
+// this commodity so they can be moved to schedulable nodes
+// Unchedulable nodes do not sell this commodity, hence pods will not be moved to these nodes.
+// We want to address the issue that pods that are already running on the unschedulable nodes are not moved out
+// due to compliance issues. This is done by not requiring the pods already on the unschedulable nodes at the time
+// of discovery to buy the 'schedulable' commodity. Without the schedulable commodity the pod continues to stay
+// on the current node unless there are resource constraints or unless moved by the kubernetes controller.
+// In addition, we need pods which do not buy the schedulable commodity to not move to other unschedulable nodes.
+// This is achieved by creating an anti-affinity policy for these pods against other unschedulable nodes.
+type UnschedulableNodesAntiAffinityGroupDTOBuilder struct {
+	cluster  *repository.ClusterSummary
+	targetId string
+	// Manager for tracking schedulable nodes
+	nodesManager *NodeSchedulabilityManager
+}
+
+func NewUnschedulableNodesAntiAffinityGroupDTOBuilder(cluster *repository.ClusterSummary,
+	targetId string, nodesManager *NodeSchedulabilityManager) *UnschedulableNodesAntiAffinityGroupDTOBuilder {
+	return &UnschedulableNodesAntiAffinityGroupDTOBuilder{
+		cluster:      cluster,
+		targetId:     targetId,
+		nodesManager: nodesManager,
+	}
+}
+
+// Creates DTOs for anti-affinity policy between the pods running on a unschedulable nodes to other unschedulable nodes.
+// This will prevent the pods running on a unschedulable nodes to be moved to other other unschedulable nodes.
+func (builder *UnschedulableNodesAntiAffinityGroupDTOBuilder) Build() []*proto.GroupDTO {
+	var policyDTOs []*proto.GroupDTO
+
+	if len(builder.nodesManager.unSchedulableNodes) <= 1 {
+		glog.Info("Zero or one unschedulable node in the cluster, anti-affinity policies will not be created")
```

Comment:
Nit: Recommend to increase verbosity level. In a lot of cases there would be just the master (in single master clusters) that will be marked unschedulable.